miden-tx/src/host/note_builder.rs

```diff
@@ -48,13 +48,14 @@ impl OutputNoteBuilder {
     ) -> Result<Self, TransactionKernelError> {
         // read note metadata info from the stack and build the metadata object
         let aux = stack[0];
-        let note_type =
-            NoteType::try_from(stack[1]).map_err(TransactionKernelError::MalformedNoteType)?;
+        let (note_type, note_execution_hint) = unmerge_type_and_hint(stack[1].into())
+            .map_err(TransactionKernelError::MalformedNoteMetadata)?;
+
         let sender =
             AccountId::try_from(stack[2]).map_err(TransactionKernelError::MalformedAccountId)?;
         let tag = NoteTag::try_from(stack[3])
             .map_err(|_| TransactionKernelError::MalformedTag(stack[3]))?;
-        let metadata = NoteMetadata::new(sender, note_type, tag, aux)
+        let metadata = NoteMetadata::new(sender, note_type, tag, note_execution_hint, aux)
             .map_err(TransactionKernelError::MalformedNoteMetadata)?;
```

Comment:
If we say that the state of the stack at the time this function is called is
```
[NOTE_METADATA, note_ptr, RECIPIENT, note_idx]
```
Then we should be able to grab the top word from the stack and convert it directly to `NoteMetadata`. In this case, we won't need to use `unmerge_type_and_hint()` function here.